aerospike/datadog_checks/aerospike/aerospike.py

```diff
@@ -259,6 +287,66 @@ def collect_datacenter(self, datacenter):
                     continue
             self.send(DATACENTER_METRIC_TYPE, key, value, datacenter_tags)
 
+    def get_metric_name(self, line):
+        # match only works at the beginning
+        # ':' or ';' are not allowed in namespace-name: https://www.aerospike.com/docs/guide/limitations.html
+        ns_metric_name_match = re.match(r'{([^\}:;]+)}-(\w+):', line)
+        if ns_metric_name_match:
+            return ns_metric_name_match.groups()[0], ns_metric_name_match.groups()[1]
+        elif line.startswith("batch-index"):
+            # https://www.aerospike.com/docs/operations/monitor/latency/#batch-index
+            return None, "batch-index"
+        else:
+            self.log.warning("Invalid data. Namespace and/or metric name not found in line: `%s`", line)
+            # Since the data come by pair and the order matters it's safer to return right away than submitting
+            # possibly wrong metrics.
+            return None, None
+
+    def collect_latencies(self, namespaces):
+        """
+        In Aerospike 5.1+, the `latencies` command is used gives the output of latencies like so:
+
+            histogramName_0:timeUnit,ops/sec,threshOld_0,threshOld_1,...;histogramName_1:...
+        """
+        data = self.get_info('latencies:')
+
+        while data:
+            line = data.pop(0)
+
+            if not data:
+                break
+
+            ns, metric_name = self.get_metric_name(line)
+            if metric_name is None:
+                return
+
+            namespace_tags = ['namespace:{}'.format(ns)] if ns else []
+            namespace_tags.extend(self._tags)
+
+            values = re.search(r'\:\w+\,(\d*\.?\d*),([,\d+.\d+]*)', line)
+            if values:
+                ops_per_sec_val = values.groups()[0]
+                # For backwards compatibility, the ops/sec value is `latencies` is already calculated
+                ops_per_sec_name = metric_name + "_" + "ops_sec"
+                self.send(NAMESPACE_LATENCY_METRIC_TYPE, ops_per_sec_name, float(ops_per_sec_val), namespace_tags)
+
+                bucket_vals = values.groups()[1]
+                if bucket_vals:
+                    latencies = bucket_vals.split(',')
+                    if latencies and len(latencies) == 17:
+                        for i in range(len(latencies)):
+                            bucket = 2 ** i
+                            tags = namespace_tags + ['bucket:{}'.format(bucket)]
+                            latency_name = metric_name
+                            self.send(NAMESPACE_LATENCY_METRIC_TYPE, latency_name, latencies[i], tags)
+
+                            # Also submit old latency names like `aerospike.namespace.latency.read_over_64ms`
+                            if bucket in [1, 8, 64]:
+                                latency_name = metric_name + '_over_{}ms'.format(str(bucket))
+                                self.send(NAMESPACE_LATENCY_METRIC_TYPE, latency_name, latencies[i], tags)
```

Comment:
should we add this new tag?

Reply:
the legacy metrics are also tagged by the `bucket:{}`